src/handlers/colonies/colonyVersionAdded.ts

```diff
@@ -2,12 +2,21 @@ import { COLONY_CURRENT_VERSION_KEY } from '~constants';
 import { ContractEvent } from '~types';
 import { toNumber, verbose } from '~utils';
 import { updateCurrentVersion } from '~utils/currentVersion';
+import { sendColonyVersionAddedNotifications } from '~utils/notifications';
 
 export default async (event: ContractEvent): Promise<void> => {
   const { version } = event.args;
   const convertedVersion = toNumber(version);
 
   verbose('New colony version:', convertedVersion, 'added to network');
 
-  await updateCurrentVersion(COLONY_CURRENT_VERSION_KEY, convertedVersion);
+  const handleVersionUpdated = async (): Promise<void> => {
+    await sendColonyVersionAddedNotifications(convertedVersion);
+  };
+
+  await updateCurrentVersion(
+    COLONY_CURRENT_VERSION_KEY,
+    convertedVersion,
+    handleVersionUpdated,
+  );
```

Comment:
is there a benefit to such callback approach over simply calling it after `updateCurrentVersion`?

Reply:
well come to think of it, mostly local dev stuff, `updateCurrentVersion` updates the version only if the "new" version is higher than the current one so I've opted for some old school callbacks :D 
Lemme know if this won't ever trigger, I can just remove it in that case and immediately invoke the function instead of passing it over
![image](https://github.com/user-attachments/assets/c777bd00-6b64-4e97-9810-a003913142be)

Reply:
How about just calling `sendColonyVersionAddedNotifications` directly inside the `updateCurrentVersion` util?

Reply:
It won't ever happen in prod or QA

> How about just calling sendColonyVersionAddedNotifications directly inside the updateCurrentVersion util?

Sounds a bit like SRP violation. The util is also used in multiple places, for different kinds of version updates

Reply:
will it happen with extensions though? because we use the same helper for updating the extension versions.
It is kinda an SRP violation, but at the same time `currentVersion` is a single entity in the database :shrug: 
Lemme know if we want to refactor this, I can do it on the extensions PR so we wrap this one up and rebase the feature branch